server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1341,6 +1342,27 @@
             sb.and("networkId", sb.entity().getNetworkId(), SearchCriteria.Op.EQ);
         }
 
+        List<Long> routersWithFailures = null;
+        if (VirtualNetworkApplianceManager.RouterHealthChecksEnabled.value() && isHealthCheckFailed != null) {
```

Comment:
@ustcweizhou what if "_router.health.checks.enabled_" config is disabled and there are last health check reports ? Will the routers listed with healthchecksfailed=trur returns empty list or notifies health checks not enabled?

Reply:
@sureshanaparti if "router.health.checks.enabled" was "true" and then changed to "false", the health check results will be out of date and incorrect. In this case, do we need to consider the health check results ?


> @ustcweizhou what if "_router.health.checks.enabled_" config is disabled and there are last health check reports ? Will the routers listed with healthchecksfailed=true returns empty list or routers with last failed reports or notifies health checks are not enabled?

Reply:
@ustcweizhou if the results are out of date/incorrect when queried for, i think better to perform health checks and update the results. (Note: if it is a sync call, then it can take time, so just notify the operator that results are out of date and perform fresh health checks before listing routers by health checks failed). See if any time duration can be set (through API or global config) to validate the results are outside of that duration or not.

Reply:
@sureshanaparti 
I think we need to make sure that the health check results are correct.
if "router.health.checks.enabled" is changed from "true" to "false", we'd better remove all existing health check results as they will not be updated any more. btw: can we perform health check via api or ui when router.health.checks.enabled" is 'false" ?

Reply:
> @sureshanaparti
> I think we need to make sure that the health check results are correct.
> if "router.health.checks.enabled" is changed from "true" to "false", we'd better remove all existing health check results as they will not be updated any more. btw: can we perform health check via api or ui when router.health.checks.enabled" is 'false" ?

@weizhouapache i think, the existing health check results are still there after changing  "router.health.checks.enabled" from "true" to "false". May be this change can be checked in existing background thread and can cleanup the old health check results.

Reply:
@sureshanaparti yes. it is another story.
I removed the check on global setting "router.health.checks.enabled" in this pr.